sbc_compassion/models/correspondence_s2b_generator.py

```diff
@@ -165,14 +165,12 @@ def preview(self):
                 )
 
                 raise UserError(msg)
-
             with Image(blob=pdf, resolution=96) as pdf_image:
                 preview = base64.b64encode(pdf_image.make_blob(format="jpeg"))
-                return self.write(
+
+                return self.isolated_write(
                     {
                         "state": "preview",
-                        "generation_status": "done",
-                        "generation_error_message": False,
```

Comment:
Shouldn't we keep this in case the preview is not called from the frontend controller but from the backend ?